Documentation/ceph-csi-snapshot.md

```diff
@@ -6,14 +6,17 @@ indent: true
 
 ## Prerequisites
 
-* Requires Kubernetes v1.17+ which supports snapshot beta.
+#### Snapshot API version support matrix
 
-* Install the new snapshot controller and snapshot beta CRD. More info can be found
-[here](https://github.com/kubernetes-csi/external-snapshotter/tree/master#usage)
+| Snapshot API version | Kubernetes Version   | Snapshot-Controller + CRDs Version | Sidecar Version |
+| -------------------- | -------------------- | ---------------------------------- | --------------- |
+| v1                   | k8s >= v1.20         | snapshot-controller >= v4.x        | sidecar >= v2.x |
+| v1beta1              | v1.17 =< k8s < v1.20 | v2.x =< snapshot-controller < v4.x | sidecar >= v2.x |
+| v1alpha1             | v1.12 =< k8s < v1.17 | snapshot-controller <= v1.x        | sidecar <= v1.x |
+Note: It is recommend to use {sidecar, controller, crds} of same version.
+- For Kubernetes v1.17+, install the snapshot controller and snapshot v1/v1beta1 CRD as required. More info can be found [here](https://github.com/kubernetes-csi/external-snapshotter/tree/master#usage).
 
-Note: If the Kubernetes distributor you are using does not supports the snapshot beta,
-still you can use the Alpha snapshots. refer to
-[snapshot](https://github.com/rook/rook/blob/release-1.3/Documentation/ceph-csi-drivers.md#rbd-snapshots)
+Note: If the Kubernetes distributor you are using does not supports the snapshot v1/v1beta1, still you can use the Alpha snapshots. refer to [snapshot](https://github.com/rook/rook/blob/release-1.3/Documentation/ceph-csi-drivers.md#rbd-snapshots)
```

Comment:
If they use alpha snapshots, they aren't compatible with v1beta1 or v1, right? We should have a warning that they will need to be discarded when they convert. I remember we had a warning about this in the past, perhaps in an upgrade guide.
```suggestion
Note: If only Alpha snapshots are available, refer to the [alpha snapshots documentation](https://github.com/rook/rook/blob/release-1.3/Documentation/ceph-csi-drivers.md#rbd-snapshots)
```

Reply:
@travisn Yes, when upgrading from snapshot beta to alpha or from alpha to v1, they will need to be discarded. 
I'll add a warning in https://github.com/rook/rook/blob/master/Documentation/ceph-upgrade.md#csi-version about snapshots and here https://github.com/rook/rook/blob/master/Documentation/ceph-upgrade.md#csi-version about sidecar version pointing back to https://github.com/rook/rook/blob/master/Documentation/ceph-csi-snapshot.md for complete description?